### PR TITLE
add a way to verify that the the license header is present in the js, ts, go files

### PR DESCRIPTION
**Pull request by Nexucis**
Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Comment by Nexucis** [comment]
the new build added should fail ... Will have to look at it a bit longer

**Comment by Nexucis** [comment]
It should help to not spend time of adding the license header where it is needed.

```bash
make checklicense
```
will verify if the license header is present where it is required

```bash
make fixlicense
```
is adding the license header where it is missing.

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,16 @@ checkformat:
...
 	! $(GOFMT) -d $$(find . -name '*.go' -print) | grep '^' ;\
 
+.PHONY: checklicense
+checklicense:
+	@echo ">> checking license"
+	./check_license.sh --check *.js *.ts *.go
+
+.PHONY: fixlicense
+fixlicense:
+	@echo ">> adding license header where it is missed"
```
Comment (juliusv) [inline Makefile:36]:
```suggestion
	@echo ">> adding license header where it's missing"
```

File: check_license.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    exit 0
+  fi
+}
+
+function add() {
+  findFileWithMissingLicense "$@" | sed 's/..$//'| while read -d $'\n' file
+  do
+    echo -e "${license_header}\n\n$(cat "${file}")" > "${file}"
+  done
+
```
Comment (juliusv) [inline check_license.sh:59]: extra empty line?

File: check_license.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+function buildPatternList() {
+  local patterns=("$@")
+  local result="-name ${patterns[0]}"
+  for pattern in "${patterns[@]:1}"; do
+    result+=" -o -name ${pattern}"
+  done
+  echo "${result}"
+}
+
+function findFileWithMissingLicense(){
```
Comment (juliusv) [inline check_license.sh:37]: `findFileWithMissingLicense` -> `findFilesWithMissingLicense`

**Comment by juliusv** [comment]
:+1:

**Pull request merged**